### ISSUE TITLE
QMimeData object error

### ISSUE THREAD

**Issue by largo84** [leo-editor/leo-editor#241]
I get the following error in the terminal window (not the log pane). It seems to happen whenever I copy and paste a block of text. It doesn't appear that I lose any data, but here it is nevertheless.

Data set on unsupported clipboard mode. QMimeData object will be deleted.

Environment: Python 3.4.4, PyQt version 5.5.1, Win10

**Comment by edreamleo** [comment]
I don't see this with:

    Python 3.5.1, PyQt version 4.8.7
    Windows 10 AMD64 (build 10.0.10586) SP0

I wonder whether this is yet another PyQt 5 problem?

**Comment by largo84** [comment]
I don't recall seeing this on my work system, which is still on PyQt4 as I recall. I'll double-check when I get back to the office.

**Comment by edreamleo** [comment]
Rev fb67b39 contains an **experimental** fix.  Tested successfully with PyQt4 (only) with Python 2 & 3 on Windows and Ubuntu.  As explained below, I have some hope that the fix will also work with PyQt5, but that remains to be proven.

@largo84 Rob, let me know what happens for you.

Ah, the power of googling...

1. Googling "Data set on unsupported clipboard mode. QMimeData object will be deleted." yields [this](https://gitlab.com/pteam/pteam-qtbase/commit/74a20b77a67ec4d5a8be0f59302075d34151dc05), so this is something new and intentional in recent Qt releases.

2. Googling "pyqt how to import qstring" yields [this](http://stackoverflow.com/questions/1400858/how-to-create-qstring-in-pyqt4). The first three answers are important, especially:

    From PyQt4 4.6+ in Python3 QString doesn't exist and
    you are supposed to use ordinary Python3 unicode objects

3. Because of these complications, Leo's Qt wrappers, in [leoQt.py](https://github.com/leo-editor/leo-editor/blob/master/leo/core/leoQt.py), now define QString regardless of imports, like this:

```
# Always define QString.
try:
    # From PyQt4 4.6+ in Python3 QString doesn't exist.
    # Use Python3 unicode objects (string literals).
    QString = QtCore.QString
except (ImportError, NameError):
    QString = g.u # g.u(s) returns s in Python 3.
```
With this in place, the relevant code in replaceClipboardWith becomes:

`cb.setText(QString(s))`

I'll leave this open until further reports come in.

**Comment by largo84** [comment]
Just a few minutes on the PC last night that has PyQt5 and I didn't see the error, so maybe it's working.

**Comment by edreamleo** [comment]
Thanks, Rob, for the confirmation.  I'll close this bug now.  Please reopen if there are further problems.

**Comment by largo84** [comment]
I'm seeing this error again and have for several days. Not sure if it was fixed or for some reason didn't see it. It shows up when I copy and paste text.

Python 3.4.3, PyQt version 5.4.1
Windows 8 AMD64 (build 6.2.9200)

**Comment by largo84** [comment]
How do I reopen this bug?

**Comment by edreamleo** [comment]
On Sun, May 15, 2016 at 6:41 PM, Rob Keeney <notifications@github.com>
wrote:

> How do I reopen this bug?
>
​If you log in there probably will be a "reopen" button on the bottom of
the page.

I just reopened it.

EKR

**Comment by edreamleo** [comment]
This error appears many times when running unit tests with PyQt5.  That should make it easier to fix.

**Comment by edreamleo** [comment]
Fixed at rev 252d1f1.

Created new LeoQtGui.setClipboardSelection method, which at present does nothing for PyQt5.
This appears to be the only way to avoid the following message:
    
    "Data set on unsupported clipboard mode. QMimeData object will be deleted."
